app/screens/convert_gm_to_channel/convert_gm_to_channel.tsx

```diff
@@ -0,0 +1,144 @@
+
+import ConvertGMToChannelForm from './convert_gm_to_channel_form';
+
+type Props = {
+    channelId: string;
+    currentUserId?: string;
+}
+
+const loadingIndicatorTimeout = 1200;
+
+const matchUserProfiles = (users: UserProfile[], members: ChannelMembership[], currentUserId: string) => {
+    // Gotta make sure we use profiles that are in members.
+    // See comment in fetchChannelMemberships for more details.
+
+    const usersById: {[id: string]: UserProfile} = {};
+    users.forEach((profile) => {
+        if (profile.id !== currentUserId) {
+            usersById[profile.id] = profile;
+        }
+    });
+
+    const filteredUsers: UserProfile[] = [];
+    members.forEach((member) => {
+        if (usersById[member.user_id]) {
+            filteredUsers.push(usersById[member.user_id]);
+        }
+    });
+
+    return filteredUsers;
+};
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        loadingContainer: {
+            justifyContent: 'center',
+            alignItems: 'center',
+            flex: 1,
+            gap: 24,
+        },
+        text: {
+            color: changeOpacity(theme.centerChannelColor, 0.56),
+            ...typography('Body', 300, 'SemiBold'),
+        },
+    };
+});
+
+const ConvertGMToChannel = ({
+    channelId,
+    currentUserId,
+}: Props) => {
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+
+    const {formatMessage} = useIntl();
+
+    const [loadingAnimationTimeout, setLoadingAnimationTimeout] = useState(false);
+    const [commonTeamsFetched, setCommonTeamsFetched] = useState(false);
+    const [channelMembersFetched, setChannelMembersFetched] = useState(false);
+    const [commonTeams, setCommonTeams] = useState<Team[]>([]);
+    const [profiles, setProfiles] = useState<UserProfile[]>([]);
+
+    const serverUrl = useServerUrl();
+    const mounted = useRef(false);
+
+    const loadingAnimationTimeoutRef = useRef<NodeJS.Timeout>();
+
+    useEffect(() => {
+        loadingAnimationTimeoutRef.current = setTimeout(() => setLoadingAnimationTimeout(true), loadingIndicatorTimeout);
+        async function work() {
+            const {teams} = await getGroupMessageMembersCommonTeams(serverUrl, channelId);
+            if (!teams || !mounted.current) {
+                return;
+            }
+            setCommonTeams(teams);
+            setCommonTeamsFetched(true);
+        }
+
+        work();
+
+        return () => {
+            clearTimeout(loadingAnimationTimeoutRef.current);
+        };
+    }, []);
+
+    useEffect(() => {
+        mounted.current = true;
+
+        return () => {
+            mounted.current = false;
+        };
+    }, []);
+
+    useEffect(() => {
+        const options: GetUsersOptions = {sort: 'admin', active: true, per_page: PER_PAGE_DEFAULT};
+        fetchChannelMemberships(serverUrl, channelId, options, true).then(({users, members}) => {
+            if (!mounted.current || !currentUserId) {
```

Comment:
`currentUserId` is a closure variable, so the value is that of when the function was created, not at the time of execution. Therefore... we can do the currentUserId check before executing the async function, since if it is defined before it is defined after, and the other way around.
```suggestion
            if (!mounted.current) {
```

Reply:
Done.